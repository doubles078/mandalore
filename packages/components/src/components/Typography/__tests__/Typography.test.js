import React from "react";
import { Typography } from "..";
import renderer from "react-test-renderer";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

test("Typography renders an h4 element for display1", () => {
  const component = renderer.create(
    <Typography variant="display1">Some text</Typography>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const typography = shallow(
    <Typography variant="display1">Some text</Typography>
  );

  expect(typography.text()).toEqual("Some text");

  expect(typography.type()).toEqual("h4");
});

test("Typography overrides variant element with component prop", () => {
  const component = renderer.create(
    <Typography variant="display1" component="p">
      Some text
    </Typography>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  const typography = shallow(
    <Typography variant="display1" component="p">
      Some text
    </Typography>
  );

  expect(typography.text()).toEqual("Some text");

  expect(typography.type()).toEqual("p");
});
