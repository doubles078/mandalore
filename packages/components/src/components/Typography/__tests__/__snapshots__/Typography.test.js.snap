// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography overrides variant element with component prop 1`] = `
<p
  className="Typography Typography--display1"
>
  Some text
</p>
`;

exports[`Typography renders an h4 element for display1 1`] = `
<h4
  className="Typography Typography--display1"
>
  Some text
</h4>
`;
